addons/stock/static/src/widgets/popover_widget.js

```diff
@@ -0,0 +1,26 @@
+/** @odoo-module */
+
+import { registry } from "@web/core/registry";
+import { _lt } from "@web/core/l10n/translation";
+
+const { Component } = owl;
+
+class jsonPpopOver extends Component {
+
+    setup() {
+        this.props = {...this.props, ...JSON.parse(this.props.value)};
+        super.setup()
+    }
+};
+
+jsonPpopOver.displayName = _lt("Json Popup");
+jsonPpopOver.supportedTypes = ["char"];
+
+class popOverLeadDays extends jsonPpopOver {};
+popOverLeadDays.template = "stock.leadDaysPopOver";
+
+class replenishmentHistoryWidget extends jsonPpopOver {};
```

Comment:
Don't forget a class start with an uppercase, and to export them in case we have to extend/patch them later
```suggestion
export class JsonPpopOver extends Component {

    setup() {
        this.props = {...this.props, ...JSON.parse(this.props.value)};
        super.setup()
    }
};

JsonPpopOver.displayName = _lt("Json Popup");
JsonPpopOver.supportedTypes = ["char"];

export class PopOverLeadDays extends jsonPpopOver {};
PopOverLeadDays.template = "stock.leadDaysPopOver";

export class ReplenishmentHistoryWidget extends jsonPpopOver {};
```